include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_reduce.h

```diff
@@ -47,24 +47,37 @@ class __reduce_mid_work_group_kernel;
 template <typename... _Name>
 class __reduce_kernel;
 
+// Adjust number of sequential operations per work-item based on the vector size. Single elements are kept to
+// improve performance of small arrays or remainder loops.
+template <std::uint8_t _VecSize, typename _Size>
+auto
+__adjust_iters_per_work_item(_Size __iters_per_work_item) -> _Size
+{
+    if (__iters_per_work_item > 1)
+        return ((__iters_per_work_item + _VecSize - 1) / _VecSize) * _VecSize;
```

Comment:
I think this can be written with `__dpl_ceiling_div` like:

```c++
return __dpl_ceiling_div(__iters_per_work_item, _VecSize) * _VecSize;
```

Reply:
Done.